In [13]:
# Imports
import pandas as pd
import altair as alt

# Loading the data
url = 'https://drive.google.com/uc?export=download&id=14O91N5OlVkvdGxXNJUj5jIsV5RexhzbB'
sessions = pd.read_csv(url)

In [48]:
# Dropping unnecessary columns
sessions_drop = sessions.drop(columns=['hashedEmail','original_start_time','original_end_time'])

In [49]:
# Creating a new column for day of the week
sessions_days = sessions_drop.assign(day_of_week = sessions_drop['start_time'].dt.dayofweek)
# Renaming the values in the day_of_week column
sessions_days['day_of_week'] = sessions_days['day_of_week'].replace({
    0:'Monday',
    1:'Tuesday',
    2:'Wednesday',
    3:'Thursday',
    4:'Friday',
    5:'Saturday',
    6:'Sunday'
})
    
sessions_days

start_time          end_time day_of_week
0    2024-06-30 18:12:00  30/06/2024 18:24      Sunday
1    2024-06-17 23:33:00  17/06/2024 23:46      Monday
2    2024-07-25 17:34:00  25/07/2024 17:57    Thursday
3    2024-07-25 03:22:00  25/07/2024 03:58    Thursday
4    2024-05-25 16:01:00  25/05/2024 16:12    Saturday
...                  ...               ...         ...
1530 2024-05-10 23:01:00  10/05/2024 23:07      Friday
1531 2024-07-01 04:08:00  01/07/2024 04:19      Monday
1532 2024-07-28 15:36:00  28/07/2024 15:57      Sunday
1533 2024-07-25 06:15:00  25/07/2024 06:22    Thursday
1534 2024-05-20 02:26:00  20/05/2024 02:45      Monday

[1535 rows x 3 columns]

### 1. Data description
#### Sessions dataset,
- Number of observations:  1535
- Number of variables/columns:  5
- Variables:

| Variable Name         | Type        | Description                                                                 |
|-----------------------|-------------|------------------------------------------------------------------------------|
| hashedEmail           | Categorical | Unique player identification                                                |
| start_time            | Temporal    | Timestamp of when player logs in (day/month/year/hour)                      |
| end_time              | Temporal    | Timestamp of when player logs off (day/month/year/hour)                     |
| original_start_time   | Temporal    | Exact second player logged on since server started                          |
| original_end_time     | Temporal    | Exact second player logged off since server started                         |
